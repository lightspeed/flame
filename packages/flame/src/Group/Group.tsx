import * as React from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { flexbox, FlexboxProps, layout, LayoutProps } from 'styled-system';
import { themeGet } from '@styled-system/theme-get';
import { Merge } from 'type-fest';

type Align = 'center' | 'right' | 'left';
type VerticalAlign = 'top' | 'bottom' | 'middle';
type GroupType = 'horizontal' | 'vertical';

export type GroupItemProps = FlexboxProps & LayoutProps;
export const GroupItem = styled('div')<GroupItemProps>`
  position: relative;
  display: flex;
  z-index: 0;

  &:hover {
    z-index: 2;
  }

  &:not(:first-of-type) {
    margin-left: ${themeGet('space.2')};
  }

  & > * {
    align-self: center;
    width: 100%;
    justify-content: center;
  }

  ${flexbox};
  ${layout};
  ${() => {
    // eslint-disable-next-line no-console
    console.warn(`
DEPRECATION WARNING
GroupItem will be removed in the next major version of Flame.
Please rely on using Flex and Box components for wrapping things properly.
    `);
    return '';
  }};
`;

// @ts-ignore
GroupItem.flameName = 'FlameGroupItem';

const horizontalAlignment = (props: { theme: any; align?: Align }) => {
  switch (props.align) {
    case 'center':
      return css`
        justify-content: center;
      `;
    case 'right':
      return css`
        justify-content: flex-end;
      `;
    case 'left':
    default:
      return css`
        justify-content: flex-start;
      `;
  }
};

const verticalAlignment = (props: { theme: any; verticalAlign?: VerticalAlign }) => {
  switch (props.verticalAlign) {
    case 'top':
      return css`
        align-items: flex-start;
      `;
    case 'bottom':
      return css`
        align-items: flex-end;
      `;
    case 'middle':
    default:
      return css`
        align-items: center;
      `;
  }
};

export type GroupAddonProps = React.HTMLAttributes<HTMLElement> &
  LayoutProps & {
    align?: Align;
    verticalAlign?: VerticalAlign;
  };
export const GroupAddon = styled('div')<GroupAddonProps>`
  display: flex;
  padding: ${themeGet('space.1')} ${themeGet('space.2')};
  text-align: center;
  background-color: ${themeGet('groupStyles.addon.background')};
  border: solid 1px ${themeGet('groupStyles.addon.borderColor')};
  border-radius: ${themeGet('radii.radius-1')};
  align-items: center;
  line-height: 1;

  ${verticalAlignment};
  ${horizontalAlignment};

  ${layout};
  ${() => {
    // eslint-disable-next-line no-console
    console.warn(`
DEPRECATION WARNING
GroupAddon will be removed in the next major version of Flame.
Please rely on InputGroup and the InputGroupAddon component.
    `);
    return '';
  }};
`;

// @ts-ignore
GroupAddon.flameName = 'FlameGroupAddon';

type BaseGroupProps = LayoutProps &
  FlexboxProps & {
    type?: GroupType;
    noSpacing?: boolean;
  };
// Due to self referential shenannigans, I'll need to disable ts on this
// @ts-ignore
const BaseGroup = styled('div')<BaseGroupProps>`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  position: relative;

  ${layout};
  ${flexbox};

  ${props =>
    !props.type &&
    !props.noSpacing &&
    css`
      /*
      Self referential shenannigans to keep same margins you'd get
      like GroupItem
    */
      & > ${BaseGroup /* eslint-disable-line */}, ${GroupAddon} {
        &:not(:first-of-type) {
          margin-left: ${themeGet('space.2')(props)};
        }
      }
    `};

  ${props =>
    props.type &&
    props.type === 'vertical' &&
    css`
      flex-direction: column;
      margin-left: 0;

      ${GroupItem} {
        flex-direction: column;
        margin: 0;

        &:not(:first-of-type) {
          margin-top: ${themeGet('space.2')(props)};
        }
      }

      ${GroupAddon} {
        padding-top: ${themeGet('space.1')(props)};
        padding-bottom: ${themeGet('space.1')(props)};
      }
    `};

  ${props =>
    props.type &&
    props.type === 'vertical' &&
    props.noSpacing &&
    css`
      flex-direction: column;

      ${GroupItem}, ${GroupAddon} {
        margin: 0;

        &:not(:first-of-type) {
          margin-top: -1px;
        }

        &:not(${GroupAddon}) * {
          border-radius: 0;
        }

        &:first-of-type,
        &:first-of-type * {
          border-top-left-radius: ${themeGet('radii.radius-1')(props)};
          border-top-right-radius: ${themeGet('radii.radius-1')(props)};
        }

        &:last-child,
        &:last-child * {
          border-bottom-left-radius: ${themeGet('radii.radius-1')(props)};
          border-bottom-right-radius: ${themeGet('radii.radius-1')(props)};
        }
        ${GroupAddon} {
          &:not(:first-of-type) {
            margin-left: -1px;
          }
        }
      }
    `};

  ${props =>
    !props.type &&
    props.noSpacing &&
    css`
      ${GroupItem}, ${GroupAddon} {
        border-radius: 0;
        &:not(:first-of-type) {
          margin-left: -1px !important;
        }

        &:not(${GroupAddon}) * {
          border-radius: 0;
        }

        &:first-of-type,
        &:first-of-type * {
          border-top-left-radius: ${themeGet('radii.radius-1')(props)};
          border-bottom-left-radius: ${themeGet('radii.radius-1')(props)};
        }

        &:last-child,
        &:last-child * {
          border-top-right-radius: ${themeGet('radii.radius-1')(props)};
          border-bottom-right-radius: ${themeGet('radii.radius-1')(props)};
        }
      }
    `};
`;

export type GroupProps = Merge<
  React.HTMLProps<HTMLElement>,
  FlexboxProps & {
    children: React.ReactNode;
    /** Sets the direction of Group. Like flexDirection. */
    type?: GroupType;
    /** Disables space between a Group's children */
    noSpacing?: boolean;
    /** Sets display: block on Group */
    block?: boolean;
    /** Sets flex=1 on Inputs when using Input inside Group */
    inputBlock?: boolean;
  }
>;
export const Group: React.FunctionComponent<GroupProps> = ({
  children,
  block,
  inputBlock,
  ...restProps
}) => {
  // eslint-disable-next-line no-console
  console.warn(`
DEPRECATION WARNING
Group will be removed in the next major version of Flame.
Please use SpacedGroup or InputGroup instead.
  `);
  const wrappedChildren = React.Children.map(children, (child: any) => {
    if (
      child &&
      child.type &&
      (child.type.flameName === 'FlameGroupItem' ||
        child.type.flameName === 'FlameGroupAddon' ||
        child.type.flameName === 'FlameGroup')
    ) {
      return child;
    }

    if (child && block) {
      return <GroupItem flex="1">{child}</GroupItem>;
    }

    // TODO: Deprecate inputBlock & flameName in favour of using GroupItem with flex
    if (child && child.type.flameName === 'Input' && inputBlock) {
      return <GroupItem flex="1">{child}</GroupItem>;
    }

    return child && <GroupItem>{child}</GroupItem>;
  });

  return <BaseGroup {...restProps}>{wrappedChildren}</BaseGroup>;
};

// @ts-ignore
Group.flameName = 'FlameGroup';
