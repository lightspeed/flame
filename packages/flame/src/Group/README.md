ðŸš¨ **Group will be deprecated in the next major version of Flame, see how each [component](#react-component) can be replaced** ðŸš¨

# Group

Group component bundles other components.

## Usage

First, make sure you have been through the [Getting started](https://github.com/lightspeed/flame#getting-started) steps of adding Flame in your application.

### React Component

#### `<Group>`

ðŸš¨ **Group will be removed in the next major version of Flame. Please use [Flex and Box](https://github.com/lightspeed/flame/tree/master/packages/flame/src/Core) or [InputGroup](https://github.com/lightspeed/flame/tree/master/packages/flame/src/InputGroup) instead.** ðŸš¨

| Prop         | Type                                | Description                                                |
| ------------ | ----------------------------------- | ---------------------------------------------------------- |
| `className`  | `string`                            | Custom className to add in addition to the default ones    |
| `children *` | `React.node`                        | The content of the group                                   |
| `type`       | `oneOf(['horizontal', 'vertical'])` | Group type (default: 'horizontal')                         |
| `noSpacing`  | `boolean`                           | Removes the default spacing between items (default: false) |
| `block`      | `boolean`                           | Children scales to parent width (default: false)           |
| `inputBlock` | `boolean`                           | Children inputs scales to parent width (default: false)    |

#### `<GroupAddon>`

ðŸš¨ **GroupAddon will be removed in the next major version of Flame. Please use [InputGroup and InputGroupAddon](https://github.com/lightspeed/flame/tree/master/packages/flame/src/InputGroup) instead.** ðŸš¨

| Prop            | Type                                 | Description                                                   |
| --------------- | ------------------------------------ | ------------------------------------------------------------- |
| `children *`    | `React.node`                         | The content of the group addon                                |
| `className`     | `string`                             | Custom className to add in addition to the default ones       |
| `align`         | `oneOf(['left', 'center', 'right'])` | Aligns the text rendered element (default: left)              |
| `verticalAlign` | `oneOf(['top', 'middle', 'bottom'])` | Vertically aligns the text rendered element (default: middle) |

#### `<GroupItem>`

ðŸš¨ **GroupItem will be removed in the next major version of Flame. Please use [Flex and Box](https://github.com/lightspeed/flame/tree/master/packages/flame/src/Core) components for wrapping things properly.** ðŸš¨

| Prop         | Type         | Description                                             |
| ------------ | ------------ | ------------------------------------------------------- |
| `children *` | `React.node` | The content of the group addon                          |
| `className`  | `string`     | Custom className to add in addition to the default ones |
| `flex`       | `string`     | Styled-system enabled flex property                     |

#### Example

```js
import React from 'react';
import { Group, GroupAddon, GroupItem } from '@lightspeed/flame/Group';
import { Button } from '@lightspeed/flame/Button';

const MyComponent = () => (
  <div>
    <Group>
      <Button>first button</Button>
      <GroupItem flex="1">
        <Button>second button</Button>
      </GroupItem>
      <Button>third button</Button>
      <GroupAddon>addon</GroupAddon>
    </Group>
  </div>
);

export default MyComponent;
```
