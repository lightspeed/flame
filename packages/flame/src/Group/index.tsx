import { Group, GroupProps, GroupAddon, GroupAddonProps, GroupItem, GroupItemProps } from './Group';

export { Group, GroupProps, GroupAddon, GroupAddonProps, GroupItem, GroupItemProps };
