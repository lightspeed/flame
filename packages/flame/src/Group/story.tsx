import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withReadme } from 'storybook-readme';

import Readme from './README.md';
import { Group, GroupAddon } from './index';
import { Input } from '../Input';
import { Box } from '../Core';
import { Alert } from '../Alert';
import { Button } from '../Button';
import { Icon } from '../Icon';
import { Text } from '../Text';
import { Badge, PillBadge } from '../Badge';

const stories = storiesOf('Components|Group', module).addDecorator(withReadme(Readme));

const Description: React.FC = ({ children }) => (
  <Text fontSize="text-s" mb={1}>
    {children}
  </Text>
);

const DeprecationWarning: React.FC = () => (
  <div className="hide-in-percy">
    <Alert
      type="warning"
      icon={<Icon name="warning" color="orange" />}
      title="Deprecation notice"
      mb={2}
    >
      Group will be deprecated in the next major version of Flame, see README on how to replace each
      of its component.
    </Alert>
  </div>
);

stories.add('Types', () => (
  <div>
    <DeprecationWarning />
    <div>
      <h3>Horizontal Group</h3>
      <Box mb={3}>
        <Group>
          <Button>-</Button>
          <Input defaultValue="0" />
          <Button>+</Button>
        </Group>
      </Box>
      <Box mb={3}>
        <Group>
          <Input placeholder="Country" />
          <Input placeholder="State" />
          <Input placeholder="City" />
        </Group>
      </Box>
      <Box mb={3}>
        <Group>
          <Button>First Button</Button>
          <Button>Second Button</Button>
          <Button>Third Button</Button>
        </Group>
      </Box>
      <Box mb={3}>
        <Group>
          <Badge type="success">Success</Badge>
          <Badge type="info">Info</Badge>
          <Badge type="danger">Danger</Badge>
          <Badge type="important">Important</Badge>
        </Group>
      </Box>
    </div>
    <div>
      <h3>Vertical Group</h3>
      <Box mb={3}>
        <Group type="vertical">
          <Button>First Button</Button>
          <Button>Second Button</Button>
          <Button>Third Button</Button>
        </Group>
      </Box>
    </div>
  </div>
));

stories.add('Spacing', () => (
  <div>
    <DeprecationWarning />
    <div>
      <h3>noSpacing Horizontal Group</h3>
      <Box mb={3}>
        <Description>
          Default <code>&lt;Group /&gt;</code>
        </Description>
        <Group noSpacing>
          <Button>-</Button>
          <Input defaultValue="0" />
          <Button>+</Button>
        </Group>
      </Box>
      <Box mb={3}>
        <Description>
          When <code>inputBlock</code> is set, the only the input stretches to its full parent width
        </Description>
        <Group noSpacing inputBlock>
          <Button>-</Button>
          <Input defaultValue="0" />
          <Button>+</Button>
        </Group>
      </Box>
      <Box mb={3}>
        <Description>
          Multiple inputs inside an <code>inputBlock</code> <code>&lt;Group /&gt;</code>
        </Description>
        <Group noSpacing inputBlock>
          <Input placeholder="Country" />
          <Input placeholder="State" />
          <Input placeholder="City" />
        </Group>
      </Box>
      <Box mb={3}>
        <Description>
          Default button <code>&lt;Group /&gt;</code>
        </Description>
        <Group noSpacing>
          <Button>First Button</Button>
          <Button>Second Button</Button>
          <Button>Third Button</Button>
        </Group>
      </Box>
      <Box mb={3}>
        <Description>
          Using <code>block</code> property scales to the parent width
        </Description>
        <Group noSpacing block>
          <Button>First Button</Button>
          <Button>Second Button</Button>
          <Button>Third Button</Button>
        </Group>
      </Box>
      <Box mb={3}>
        <Description>
          Creating <code>&lt;Group /&gt;</code> inside <code>&lt;Group /&gt;</code>
        </Description>
        <Group>
          <Group noSpacing>
            <Button>Button</Button>
            <Button>Button</Button>
          </Group>
          <Group noSpacing inputBlock>
            <Button>Button</Button>
            <GroupAddon>
              <Icon name="ecom-am" baseColor="blue" detailsColor="green" />
            </GroupAddon>
            <Input placeholder="Country" />
            <Button>Button</Button>
          </Group>
        </Group>
      </Box>
      <Box mb={3}>
        <Description>
          Text vertical aligns inside <code>&lt;Group /&gt;</code> components
        </Description>
        <Group>
          <div>Some text beside a button</div>
          <Button>First Button</Button>
        </Group>
      </Box>
    </div>
    <div>
      <h3>noSpacing Vertical Group</h3>
      <p>
        Note when using the vertical group, you will probably need to pass{' '}
        {<code>align={'"center"'}</code>} yourself to make sure {"it's"} aligned correctly
      </p>
      <Group type="vertical" noSpacing>
        <GroupAddon align="center">Header Addon</GroupAddon>
        <Button>First Button</Button>
        <Button>Second Button</Button>
        <Button>Third Button</Button>
        <GroupAddon align="center">Footer Addon</GroupAddon>
      </Group>
    </div>
  </div>
));

stories.add('Addon', () => (
  <div>
    <DeprecationWarning />
    <div>
      <h3>Horizontal Group Addon</h3>
      <Box mb={3}>
        <Group>
          <GroupAddon>
            <Icon name="customers" />
          </GroupAddon>
          <Input placeholder="Group addons" />
          <GroupAddon>
            / cm<sup>2</sup>
          </GroupAddon>
        </Group>
      </Box>

      <Box mb={3}>
        <Group noSpacing>
          <GroupAddon>$</GroupAddon>
          <Input placeholder="Group addons" />
          <GroupAddon>
            / cm<sup>2</sup>
          </GroupAddon>
        </Group>
      </Box>

      <h4>Addon horizontal positioning</h4>
      <Text as="h5" mb={1}>
        align={'"left"'}
        (Default)
      </Text>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon align="left" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            <Input />
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        align={'"center"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon align="center" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            <Input />
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        align={'"right"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon align="right" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            <Input />
          </Group>
        </div>
      </Box>

      <h4>Addon vertically positioning</h4>
      <Text as="h5" mb={1}>
        verticalAlign={'"top"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon verticalAlign="top" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            {/* TODO: make use of flame-textarea when it arrives */}
            <textarea rows={5} />
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        verticalAlign={'"middle"'} (default)
      </Text>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon verticalAlign="middle" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            {/* TODO: make use of flame-textarea when it arrives */}
            <textarea rows={5} />
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        verticalAlign={'"bottom"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon verticalAlign="bottom" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            {/* TODO: make use of flame-textarea when it arrives */}
            <textarea rows={5} />
          </Group>
        </div>
      </Box>

      <h4>Custom classes</h4>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon className="width-200px">Fixed width of 200px</GroupAddon>
            <Input placeholder="Group addons" />
          </Group>
        </div>
      </Box>

      <Box mb={3}>
        <Group noSpacing block>
          <GroupAddon className="width-200px overflow-hidden">
            <div className="ellipsis">Example using ellipsis thats very long</div>
          </GroupAddon>
          <Input placeholder="Group addons" />
        </Group>
      </Box>

      <h4>Fixed width and {'align="center"'}</h4>
      <Box mb={3}>
        <div>
          <Group noSpacing block>
            <GroupAddon align="center" className="width-200px">
              Fixed width of 200px
            </GroupAddon>
            <Input placeholder="Group addons" />
          </Group>
        </div>
      </Box>
    </div>

    <div>
      <h3>Vertical Group Addon</h3>
      <p>
        Note when using the vertical group, you will probably need to pass{' '}
        {<code>align={'"center"'}</code>} yourself to make sure {"it's"} aligned correctly
      </p>
      <Box mb={3}>
        <Box mb={3}>
          <Group type="vertical" noSpacing>
            <Button>First Button</Button>
            <GroupAddon align="center">
              <Icon name="dashboard" />
            </GroupAddon>
            <Input placeholder="State" />
            <GroupAddon align="center">
              <PillBadge type="important" size="small">
                Important
              </PillBadge>
            </GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </Box>
      </Box>

      <h4>Addon horizontal positioning</h4>
      <Text as="h5" mb={1}>
        align={'"left"'} (Default)
      </Text>
      <Box mb={3}>
        <div>
          <Group type="vertical" noSpacing block>
            <Button>First Button</Button>
            <GroupAddon align="left">Fixed width of 200px</GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        align={'"center"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group type="vertical" noSpacing block>
            <Button>First Button</Button>
            <GroupAddon align="center">Fixed width of 200px</GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        align={'"right"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group type="vertical" noSpacing block>
            <Button>First Button</Button>
            <GroupAddon align="right">Fixed width of 200px</GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </div>
      </Box>

      <h4>Addon vertically positioning</h4>
      <Text as="h5" mb={1}>
        verticalAlign={'"top"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group type="vertical" noSpacing block>
            <Button>First Button</Button>
            <GroupAddon verticalAlign="top" className="height-100px">
              Fixed height of 100px
            </GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        verticalAlign={'"middle"'} (default)
      </Text>
      <Box mb={3}>
        <div>
          <Group type="vertical" noSpacing block>
            <Button>First Button</Button>
            <GroupAddon verticalAlign="middle" className="height-100px">
              Fixed height of 100px
            </GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </div>
      </Box>

      <Text as="h5" mb={1}>
        verticalAlign={'"bottom"'}
      </Text>
      <Box mb={3}>
        <div>
          <Group type="vertical" noSpacing block>
            <Button>First Button</Button>
            <GroupAddon verticalAlign="bottom" className="height-100px">
              Fixed height of 100px
            </GroupAddon>
            <Button>Second Button</Button>
          </Group>
        </div>
      </Box>
    </div>
  </div>
));
