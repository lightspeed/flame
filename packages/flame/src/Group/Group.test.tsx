/* eslint-disable import/no-extraneous-dependencies */

import * as React from 'react';
import { createComponent, customRender } from 'test-utils';
import { Group, GroupAddon } from './Group';

describe('<Group />', () => {
  describe('Snapshots', () => {
    it('renders a group', () => {
      const component = createComponent(
        <Group>
          <GroupAddon>$</GroupAddon>
          <input type="text" />
          <button type="button">My button</button>
          <GroupAddon>/ per unit</GroupAddon>
        </Group>,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('Render', () => {
    it('render Group with a className attribute', () => {
      const className = 'custom-class';
      const groupAddonClassName = 'groupAddonClassName';
      const anotherClassName = 'anotherClassName';

      const { container } = customRender(
        <Group className={className}>
          <GroupAddon className={groupAddonClassName}>$</GroupAddon>
          <span className={anotherClassName}>Test Group</span>
        </Group>,
      );
      const groupEl = container.querySelector('.custom-class');
      const groupAddonEl = container.querySelector('.groupAddonClassName');
      const spanEl = container.querySelector('.anotherClassName');

      expect(groupEl).toBeTruthy();
      expect(groupAddonEl).toBeTruthy();
      expect(spanEl).toBeTruthy();
    });
  });
});
