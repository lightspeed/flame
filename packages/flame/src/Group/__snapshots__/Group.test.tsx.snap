// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Group /> Snapshots renders a group 1`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: relative;
}

.emotion-8 > .emotion-9:not(:first-of-type),
.emotion-8 .emotion-1:not(:first-of-type) {
  margin-left: 0.75rem;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.375rem 0.75rem;
  text-align: center;
  background-color: #f3f3f3;
  border: solid 1px #abb3b3;
  border-radius: 0.1875rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 0;
}

.emotion-2:hover {
  z-index: 2;
}

.emotion-2:not(:first-of-type) {
  margin-left: 0.75rem;
}

.emotion-2 > * {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-0 emotion-1"
  >
    $
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <input
      type="text"
    />
  </div>
  <div
    className="emotion-2 emotion-3"
  >
    <button
      type="button"
    >
      My button
    </button>
  </div>
  <div
    className="emotion-0 emotion-1"
  >
    / per unit
  </div>
</div>
`;
